Gutenboarding: Fix theme name + premium badge center alignment in design selection

ciampo opened the pull request:
#### Changes proposed in this Pull Request

* Changed the layout of the `.design-selector__option-meta` element from `inline-grid` to `inline-flex`
* This allows for its content to be always horizontally centered, regardless of the theme being premium or not
* Theme name and premium label allowed to wrap on more lines, in case they don't fit into one line
* Added the `font-family: inherit` rule on the `.design-selector__design-option` button element, in order to prevent User Agent stylesheet from overriding the font family used for the theme name and premium badge
* Other typography-related adjustments:
  - Smaller font size for theme name
  - Smaller font size for premium badge. _(Note: it's now 10px, which is not one of the [allowed font sizes](https://github.com/Automattic/wp-calypso/blob/master/packages/typography/styles/variables.scss) — I've disabled the linter rule for now, and added the `//typography-exceptioninline` comment)_
  - Smaller icon size for premium badge
  - DIfferent background for the Premium badge (gray 80), instead of pure black
  - Uppercase text in the premium badge


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* `yarn start`
* visit `localhost:3000/new`
* proceed through the onboarding flow until reaching the Design (theme) selection step
* scroll through the page until premium themes are shown
* notice how both non-premium and premium themes have their name center-aligned
* notice how the font used for the theme name and the premium badge is now aligned to the rest of the UI
* notice how the premium badge has now uppercase text and a dark gray background (instead of black


#### Before

![Theme names and premium label are not horizontally centered](https://user-images.githubusercontent.com/1083581/91046846-6ddd1e00-e619-11ea-9736-e2d5af390e47.png)


#### After

![Theme name and badge are centered](https://user-images.githubusercontent.com/1083581/91146819-b3502880-e6b7-11ea-9a9c-01c09f95c6fd.png)


Fixes #44685

yansern commented:
Tested on Chrome & also tested on RTL on Firefox.
![2020-08-24_16-06-47](https://user-images.githubusercontent.com/1287077/91053911-d4673980-e623-11ea-8bf8-e8dcc88e7a12.png)

yansern commented:
Perhaps the font size for the badge could be smaller?

![image](https://user-images.githubusercontent.com/1287077/91054411-0e384000-e624-11ea-8c16-121360ea46a8.png)



This is how it looks like in Figma. (Also it seems that it is full caps).

![image](https://user-images.githubusercontent.com/1287077/91054894-332cb300-e624-11ea-9b97-290c9ae989d9.png)

yansern commented on client/landing/gutenboarding/onboarding-block/design-selector/style.scss:
```diff
@@ -107,11 +107,12 @@
-		display: inline-grid;
+		align-items: center;
+		display: inline-flex;
+		flex-wrap: wrap; // If theme name and premium badge don't fit on one line
```
Very useful comment! Thanks!

yansern reviewed (changes requested):
Perhaps the font size for the badge could be smaller?

![image](https://user-images.githubusercontent.com/1287077/91054411-0e384000-e624-11ea-8c16-121360ea46a8.png)



This is how it looks like in Figma. (Also it seems that it is full caps).

![image](https://user-images.githubusercontent.com/1287077/91054894-332cb300-e624-11ea-9b97-290c9ae989d9.png)

ciampo commented:
Updated theme name & premium badge typography, following the Figma spec:

- Smaller font size for theme name
- Smaller font size for premium badge (note: it's now 10px, which causes a lint error as that is not one of the allowed font sizes — I've disabled the linter rule for now, and added the `//typography-exception`inline comment)
- Smaller icon size for premium badge
- DIfferent background for the Premium badge (gray 80), instead of pure black
- Uppercase text in the premium badge

razvanpapadopol commented:
> Smaller font size for premium badge. Note: it's now 10px, which causes a lint error as that is not one of the allowed font sizes (see packages/typography/styles/variables.scss) — I've disabled the linter rule for now, and added the //typography-exceptioninline comment

Not a blocker for this PR but we should take this example and some bigger font-size we need for the PlansGrid and try updating these rules and [typography variables](https://github.com/Automattic/wp-calypso/blob/master/packages/typography/styles/variables.scss). 

Small vertical alignment issue between design name and badge:
<img width="339" alt="Screenshot 2020-08-25 at 09 57 40" src="https://user-images.githubusercontent.com/14192054/91143063-98cb7e80-e6ba-11ea-8b96-747fcb2749fa.png">

Note: we should remember to update also the **After** screenshot in description once the implementation is final (or at least ready for review).

ciampo commented:
> Not a blocker for this PR but we should take this example and a similar one for the PlansGrid and propose an update for this rule and typography variables.

Is there anything I can help with this?


> There is only some vertical alignment issue between design name and badge:

I adjusted the vertical alignment to the best I could before starting to fight with sub-pixel alignment


> Note: we should remember to update also the After screenshot in description once the implementation is final (or at least ready for review).

Good point — it's done now

yansern commented:
It appears that the tooltip becomes uppercase as well (but we can leave that for the other PR).
![image](https://user-images.githubusercontent.com/1287077/91150962-56577100-e6bd-11ea-9c20-05b5f6c3edcf.png)

ciampo commented:
Rebased on top of master

yansern reviewed (approved):
Approving this PR. Other minor tweaks can be pushed to another PR.
